misc/install_initrd.sh

```diff
@@ -0,0 +1,245 @@
+        sed -i 's,BOOTDEVICE,'"${boot_device}"',' /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh
+        echo " - chmod +x /usr/lib/dracut/modules.d/96rapiddisk/* ..."
+        chmod +x /usr/lib/dracut/modules.d/96rapiddisk/*
+    else
+        echo " - Uninstalling by removing dir /usr/lib/dracut/modules.d/96rapiddisk..."
+        if [ ! -d /usr/lib/dracut/modules.d/96rapiddisk ] && [ -z $force ] ; then
+            myerror "Error: /usr/lib/dracut/modules.d/96rapiddisk does not exist, use '--force'. Exiting..."
+        fi
+        rm -rf /usr/lib/dracut/modules.d/96rapiddisk
+    fi
+
+    echo " - Running 'dracut --kver $kernel_version -f'"
+    #dracut --add-drivers "rapiddisk rapiddisk-cache" --kver "$kernel_version" -f
+    dracut --kver "$kernel_version" -f
+
+    echo " - Done under CentOS. A reboot is needed."
+    exit 0
+
+elif hostnamectl | grep "Ubuntu" > /dev/null 2> /dev/null; then
+
+    # Ubuntu
+    echo " - Ubuntu detected!"
+
+    # These checks are intended to find the best place to put the scripts under Ubuntu
+    if [ -d /etc/initramfs-tools/hooks ] && [ -d /etc/initramfs-tools/scripts/init-premount ]; then
+        hook_dest=/etc/initramfs-tools/hooks/rapiddisk_hook
+        bootscript_dest=/etc/initramfs-tools/scripts/init-premount/rapiddisk
+    elif [ -d /usr/share/initramfs-tools/hooks ] && [ -d /usr/share/initramfs-tools/scripts/init-premount ]; then
+        hook_dest=/usr/share/initramfs-tools/hooks/rapiddisk_hook
+        bootscript_dest=/usr/share/initramfs-tools/scripts/init-premount/rapiddisk
+    else
+        myerror "Error: I can't find any suitable place for initramfs scripts. Exiting..."
+    fi
+
+    if [ -n "$uninstall" ]; then
+        echo " - Starting uninstall process..."
+
+        if [ -x "$hook_dest" ] || [ -x "$bootscript_dest" ] && [ -z $force ]; then
+            echo " - Uninstalling scripts..."
+            rm -f "$hook_dest" 2> /dev/null
+            rm -f "$bootscript_dest" 2> /dev/null
+        else
+            myerror "Error: hook and/or boot scripts are not installed, use '--force'. Exiting..."
+        fi
+
+        # TODO this check is not much useful since update-initrd includes many modules automatically and can lead to false
+        #  positves
+
+        #if lsinitramfs /boot/initrd.img-"$kernel_version" | grep rapiddisk >/dev/null 2>/dev/null ; then
+        #    echo ""
+        #    echo "Error: /boot/initrd.img-$kernel_version still contains rapiddisk stuff."
+        #    echo
+        #    exit 1
+        #fi
+
+        finalstuff
+        exit 0
+    fi
+
+    ##############################################################################
+    # TODO The following piece of code was there to check if the user specified
+    # the same device present in /etc/fstab - but if in fstab UUIDs are used, or
+    # the LABEL= syntax, it loses meaning.
+    ##############################################################################
```

Comment:
If in fstab the device is specified via UUID or LABEL it is possible to check `/dev/disk/by-*` to "resolve" it into the classic `/dev/xxxx` syntx... I don't know if it really makes sense.

Reply:
I think most modern distros tend to use the UUID approach. But if we initially label this as experimental and have a README.md highlighting features yet-to-be integrated, we should be good for an initial merge.